pkg/client/client_test.go

```diff
@@ -813,5 +814,7 @@ func addToScheme(t *testing.T) *runtime.Scheme {
 	require.NoError(t, err)
 	err = toolchainv1alpha1.AddToScheme(s)
 	require.NoError(t, err)
+	err = templatev1.Install(s)
```

Comment:
Out of curiosity, why does Install need to be called now, whereas before it was not?

Reply:
Why do we need to call the `Install` here? The answer is simple - because one of the tests processes a template content into a `Template` object. More interesting is the second part of your question - why didn't we have to call it in the previous code. 

TBH with you - this scheme is a kind of magic black box for me, so I'm not sure. There were some changes in the default set of the API schemes that are automatically added to that instance `scheme.Scheme`, but the template is not one of them. There were also some changes in the conversion logic that converts the parsed `Unstructured` objects into the typed object, but I didn't dig so much deep into it. 
I also printed out (in the original code) the content of the scheme before it is passed to the `Convert` function and the template API scheme is not there, so I have no clue how it could work. My guess is that the original conversion code from the apimachinery library did some default assumption or added the template scheme there - but as I said - I'm not sure :man_shrugging:

Reply:
Thanks for taking the time to explain it so thoroughly